src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_inlet_node_name").get<std::string>());
+                std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_outlet_node_name").get<std::string>());
+                // Implicit
+                std::string condenserType = "AIRSOURCE"; // UtilityRoutines::MakeUPPERCase(fields.at("condenser_type").get<std::string>());
+                // A4
+                std::string sourceSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("air_source_node_name").get<std::string>());
+                std::string sourceSideOutletNodeName =
+                    "DUMMY_CONDENSER"; // UtilityRoutines::MakeUPPERCase(fields.at("source_side_outlet_node_name").get<std::string>());
+                srand(time(NULL));
+                for (int i = 1; i <= 3; ++i) {
+                    sourceSideOutletNodeName += ("_" + std::to_string(rand()));
+                }
+
+                // A5
+                if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) {
+                    if (fields.find("companion_cooling_heat_pump_name") != fields.end()) { // optional field
+                        thisPLHP.companionCoilName = UtilityRoutines::MakeUPPERCase(fields.at("companion_cooling_heat_pump_name").get<std::string>());
+                    }
+                } else if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling) {
+                    if (fields.find("companion_heating_heat_pump_name") != fields.end()) { // optional field
+                        thisPLHP.companionCoilName = UtilityRoutines::MakeUPPERCase(fields.at("companion_heating_heat_pump_name").get<std::string>());
+                    }
+                }
+
+                // A6 Fuel Type
+                thisPLHP.fuelTypeStr = UtilityRoutines::MakeUPPERCase(fields.at("fuel_type").get<std::string>());
+                // Validate fuel type input
+                // Locals
+                static constexpr std::string_view RoutineName("processInputForEIRPLHP: ");
+                bool FuelTypeError(false);
+                UtilityRoutines::ValidateFuelTypeWithAssignResourceTypeNum(
+                    thisPLHP.fuelTypeStr, thisPLHP.GAHPFuelTypeForOutputVariable, thisPLHP.fuelType, FuelTypeError);
+                if (FuelTypeError) {
+                    ShowSevereError(state, fmt::format("{}{}=\"{}\",", RoutineName, cCurrentModuleObject, thisPLHP.name));
+                    // ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(2) + '=' + state.dataIPShortCut->cAlphaArgs(2));
+                    ShowContinueError(state, "Invalid Fuel Type = " + thisPLHP.fuelTypeStr);
+                    // Set to Electric to avoid errors when setting up output variables
+                    thisPLHP.GAHPFuelTypeForOutputVariable = "NaturalGas";
```

Comment:
Can look this up when you need to, don't need to store it.

Reply:
Addressed in earlier changes.